packages/core/core/src/services/entity-validator/validators.ts

```diff
@@ -163,13 +166,20 @@ const addUniqueValidator = <T extends strapiUtils.yup.AnySchema>(
       return true;
     }
 
-    const whereParams = entity
-      ? { $and: [{ [updatedAttribute.name]: value }, { $not: { id: entity.id } }] }
+    /**
+     * At this point we know that we are creating a new entry or that the unique field value has changed
+     * We check if there is an entry of this content type in the same locale with the same unique field value
+     */
+    const filters = options.locale
+      ? {
+          [updatedAttribute.name]: value,
+          locale: options.locale,
+        }
       : { [updatedAttribute.name]: value };
 
-    const record = await strapi.query(model.uid).findOne({
-      select: ['id'],
-      where: whereParams,
+    const record = await strapi.documents.findFirst(model.uid, {
+      filters,
+      status: options.isDraft ? 'draft' : 'published',
```

Comment:
Preferably, the locale should be placed as a top level attribute :
```suggestion
    const record = await strapi.documents.findFirst(model.uid, {
      filters: { [updatedAttribute.name]: value },
      locale: options.locale
      status: options.isDraft ? 'draft' : 'published',
```